framework/src/org/checkerframework/framework/type/TypesIntoElements.java

```diff
@@ -244,7 +244,8 @@ private static void addUniqueTypeCompounds(Types types, Symbol sym, List<TypeCom
         List<Attribute.TypeCompound> res = List.nil();
 
         for (Attribute.TypeCompound tc : tcs) {
-            if (!TypeAnnotationUtils.isTypeCompoundContained(types, raw, tc)) {
+            // Don't insert an annotation if one with the same name already exists.
+            if (!TypeAnnotationUtils.isTypeCompoundContainedByName(types, raw, tc)) {
```

Comment:
Would it make sense to replace an existing annotation with the canonical version?
Let's say the programmer wrote `@A("foo")` which get's normalized to `@A("this.foo")` in the ATM.
However, this change seems to leave the annotation in the Element as `@A("foo")`.
Wouldn't it be better to find existing `@A` annotations and replace them with the canonical version?
Aren't you afraid of inconsistent results, in particular when looking at Elements for Trees that might have been thrown out?